File: quarkus-tmdb/deployment/src/test/java/dev/reuss/tmdb/quarkus/deployment/TmdbMetricsProcessorTest.java
package dev.reuss.tmdb.quarkus.deployment;

import io.quarkus.deployment.Capabilities;
import io.quarkus.deployment.Capability;
import org.junit.jupiter.api.Test;

import java.util.Set;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

class TmdbMetricsProcessorTest {

    @Test
    void shouldRegisterMetricsWhenMetricsCapabilityIsAvailable() {
        Capabilities capabilities = new Capabilities(Set.of(Capability.METRICS));

        assertTrue(TmdbMetricsProcessor.shouldRegisterMetrics(capabilities));
    }

    @Test
    void shouldSkipMetricsWhenMetricsCapabilityIsMissing() {
        Capabilities capabilities = new Capabilities(Set.of());

        assertFalse(TmdbMetricsProcessor.shouldRegisterMetrics(capabilities));
    }
}

File: quarkus-tmdb/deployment/src/test/kotlin/dev/reuss/tmdb/quarkus/deployment/TmdbMetricsProcessorTest.kt
package dev.reuss.tmdb.quarkus.deployment

import io.quarkus.deployment.Capabilities
import io.quarkus.deployment.Capability
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

class TmdbMetricsProcessorTest {
    @Test
    fun shouldRegisterMetricsWhenMetricsCapabilityIsAvailable() {
        val capabilities =
            Capabilities(
                setOf(Capability.METRICS),
            )

        assertTrue(
            TmdbMetricsProcessor.shouldRegisterMetrics(capabilities),
        )
    }

    @Test
    fun shouldSkipMetricsWhenMetricsCapabilityIsMissing() {
        val capabilities =
            Capabilities(
                emptySet(),
            )

        assertFalse(
            TmdbMetricsProcessor.shouldRegisterMetrics(capabilities),
        )
    }
}
